src/components/App/App.js

```diff
@@ -6,66 +6,300 @@ import Movies from "../Movies/Movies";
 import SavedMovies from "../SavedMovies/SavedMovies";
 import Profile from "../Profile/Profile";
 import { Register } from "../Register/Register";
-import { Routes, Route, useNavigate } from "react-router-dom";
+import {
+  Routes,
+  Route,
+  useNavigate,
+  useLocation,
+  Navigate,
+} from "react-router-dom";
 import { Login } from "../Login/Login";
 import NotFound from "../NotFound/NotFound";
+import Preloader from "../Preloader/Preloader.js";
+import { CurrentUserContext } from "../../context/CurrentUserContext.js";
+import * as auth from "../../utils/MainApi.js";
+import api from "../../utils/MoviesApi.js";
+import ProtectedRouteElement from "../ProtectedRoute/ProtectedRoute.js";
+import {
+  DuplicateError,
+  RegisterError,
+  ServerError,
+  LoginError,
+  ProfileError,
+  LoadingFilmError,
+  LoadingProfileSuccess,
+} from "../../utils/constants.js";
 
 function App() {
-  //поменять значение на false чтобы разлогиниться
-  //роуты пока не защищены
-  const [loggedIn, setLoggedIn] = useState(true);
+  const [loggedIn, setLoggedIn] = useState(localStorage.getItem("token"));
+  const [loading, setLoading] = useState(false);
+  const [currentUser, setCurrentUser] = useState("");
+  const [movies, setMovies] = useState([]);
+  const [savedMovies, setSavedMovies] = useState([]);
+  const [errorRegister, setErrorRegister] = useState(null);
+  const [errorLogin, setErrorLogin] = useState(null);
+  const [errorProfile, setErrorProfile] = useState(null);
+  const [successProfile, setSuccessProfile] = useState(null);
+  const [errorLoadingMovies, setErrorLoadingMovies] = useState(null);
+  const navigate = useNavigate();
+  const location = useLocation();
+  const isSignUp = location.pathname === "/signup";
+  const isSignIn = location.pathname === "/signin";
 
-  //поменять значение на null чтобы не видеть сообщения об ошибке в Profile
-  const [error, setError] = useState("Пользователь уже существует");
+  useEffect(() => {
+    if (!isSignUp) {
+      setErrorRegister(null);
+    }
+    if (!isSignIn) setErrorLogin(null);
+  }, [isSignIn, isSignUp]);
 
-  //const navigate = useNavigate();
+  useEffect(() => {
+    handleTokenCheck();
+  }, [loggedIn]);
+
+  const handleTokenCheck = () => {
+    const jwt = localStorage.getItem("token");
+    if (jwt) {
+      auth
+        .checkToken(jwt)
+        .then((res) => {
+          setCurrentUser(res);
+        })
+        .catch((error) => {
+          console.log(`Ошибка при проверке токена ${error}`);
+          setLoggedIn(false);
+          navigate("/signin", { replace: true });
+        });
+    }
+  };
+
+  useEffect(() => {
+    if (loggedIn) {
+      Promise.all([auth.getItems(), auth.getUser()])
```

Comment:
why did you call Axios in the main file? You can create separate Axios client, endpoint files containing all APIs endpoints. Use react query for better understanding and easy-to-handle.